engine/src/execution_graph/logic_controllers/BatchOrderByProcessing.h

```diff
@@ -137,6 +138,20 @@ std::size_t PARTITION_PLAN_MESSAGE_TRACKER_IDX = 1;
 				}
 				
 				total_table_rows.push_back(local_total_num_rows);
+				total_avg_bytes_per_row.push_back(avg_bytes_per_row);
+
+				// let's recompute the avg_bytes_per_row if node master does not have data
+				if (avg_bytes_per_row == 1) {
```

Comment:
I think we should always use that info, not only if the master node does not have it.
Second, i think you should weight the avg_bytes_per_row info by the amount of data from that node, such that if one node has little data but has a very high or low avg_bytes_per_row, it does not skew the results. So I would recommend doing something like:
```
size_t total_bytes = 0; 
size_t total_rows = 0;
for(std::size_t i = 0; i < nodes.size(); ++i) {
total_bytes += total_avg_bytes_per_row[i]*total_table_rows[i];
total_rows += total_table_rows[i];
}
size_t final_avg_bytes_per_row = total_bytes/total_rows;

Reply:
and use final_avg_bytes_per_row for generate_partition_plan

Reply:
updated as requested.